### ISSUE TITLE
Do not exceed upper bound of range when using --target semver

### ISSUE THREAD

**Issue by raineorshine** [raineorshine/npm-check-updates#1409]
This is very useful for keeping my packages up to date, but often I have to set a ceiling version number due to newer versions dropping support for Node versions. However this then makes this tool not useful at all as it ignores that constraint and tells me to update past that which I am incapable of doing.

I want the command to take that ceiling into account and only offer me upgrades that are below the ceiling I have set.
Upgrading past the ceiling point leaves me unable to see when there are genuine patch or minor upgrades etc that I would like to upgrade to.

Example current usage:
```
 binary-extensions       ^2.2.0  →  ^3.0.0
 commander           ^9.5.0 <10  →     ^12
 cross-fetch          ^3.1.8 <4  →      ^4
 isbinaryfile        ^4.0.10 <5  →      ^5
 typescript         ~5.0.4 <5.1  →    ^5.4
```

I thought `ncu --target semver` would do what I like but no.

Is there any usage that does do what I wish it to do?

Thanks

_Originally posted by @Nixinova in https://github.com/raineorshine/npm-check-updates/discussions/1407_

**Comment by raineorshine** [comment]
@Nixinova @XhmikosR Is it preferable to upgrade the lower bound or simply skip the dependency?

**Comment by XhmikosR** [comment]
I don't think it makes sense to bump the lower bound. The whole idea is to bump to the newest version that is still satisfiable by semver.

**Comment by XhmikosR** [comment]
Unless you mean something else that I might not getting :/

**Comment by raineorshine** [comment]
https://github.com/raineorshine/npm-check-updates/pull/1920/changes#diff-d6fcf7a3f499e0536e86de96c2cf3e0e828c1b2e079a0691a9ba553bea3a2c5cR387 upgrades the lower bound only. I just wanted to check and see if that was the desired behavior, or if it would be better to skip the package entirely.

**Comment by XhmikosR** [comment]
Sorry, I confused myself with the wording :/.

Bumping the lower bound is what I personally think is better. For a caret range the lower bound is the target version, so `^1.0.0` -> `^1.2.0` is already what --target semver does. `^1.0.0 <2` -> `^1.2.0 <2` is the same thing but keeps the `<2` ceiling.

Skipping would be inconsistent since `^1.0.0` and `^1.0.0 <2` are basically the same range. And it only bumps when a newer version actually fits under the ceiling, so it never goes past it.

**Comment by raineorshine** [comment]
Good point. Thanks for breaking it down that way.

**Comment by Nixinova** [comment]
Thanks!
